testsuites/CBLTester/system_tests/test_system_concurrent_testing.py

```diff
@@ -0,0 +1,384 @@
+
+    db_name = cbl_params["db_name"]
+    db_obj = cbl_params["db_obj"]
+    cbl_db = cbl_params["cbl_db"]
+
+    sg_channels = sg_params["sg_channels"]
+
+    thread_name = "Thread {}".format(db_name)
+    func_name = "cbl_db_init"
+
+    _log_system_test(thread_name, func_name, "Adding doc on {} db".format(db_name))
+    doc_id_set = set()
+    doc_prefix = "{}_doc".format(db_name)
+    _log_system_test(thread_name, func_name, doc_prefix)
+    _log_system_test(thread_name, func_name, num_of_itr_per_db)
+    # adding max docs per batch
+    for i in range(num_of_itr_per_db):
+        ids = db_obj.create_bulk_docs(num_of_docs_in_itr, doc_prefix, db=cbl_db, channels=sg_channels,
+                                      id_start_num=i * num_of_docs_in_itr, generator=generator)
+        _log_system_test(thread_name, func_name, db_name)
+        _log_system_test(thread_name, func_name, ids)
+        doc_id_set.update(ids)
+    # adding remaining docs
+    _log_system_test(thread_name, func_name, "extra")
+    _log_system_test(thread_name, func_name, extra_docs_in_itr_per_db)
+    if extra_docs_in_itr_per_db != 0:
+        ids = db_obj.create_bulk_docs(extra_docs_in_itr_per_db, doc_prefix, db=cbl_db, channels=sg_channels,
+                                      id_start_num=(i + 1) * num_of_docs_in_itr, generator=generator)
+        _log_system_test(thread_name, func_name, ids)
+        doc_id_set.update(ids)
+
+    return {db_name: doc_id_set}
+
+
+def process_per_cbl_client(sg_params, cbl_params, test_params, doc_ids):
+    # run test per cbl db from here
+    query_limit = 1000
+    query_offset = 0
+
+    # pulling test specific parameters
+    up_time = test_params["up_time"]
+    repl_status_check_sleep_time = test_params["repl_status_check_sleep_time"]
+    num_of_docs_to_update = test_params["num_of_docs_to_update"]
+    num_of_docs_to_delete = test_params["num_of_docs_to_delete"]
+    num_of_docs_to_add = test_params["num_of_docs_to_add"]
+    doc_id_for_new_docs = test_params["num_of_docs"]
+    num_of_doc_updates = test_params["num_of_doc_updates"]
+    generator = test_params["generator"]
+
+    # pulling sync gateway paramters
+    sg_db = sg_params["sg_db"]
+    sg_admin_url = sg_params["sg_admin_url"]
+    sg_blip_url = sg_params["sg_blip_url"]
+    sg_url = sg_params["sg_url"]
+    sg_client = sg_params["sg_client"]
+    sg_channels = sg_params["sg_channels"]
+    username = sg_params["username"]
+
+    # pulling couchbase lite parameters
+    base_url = cbl_params["base_url"]
+    cbl_db = cbl_params["cbl_db"]
+    query = cbl_params["query"]
+    db_name = cbl_params["db_name"]
+    db_obj = cbl_params["db_obj"]
+    platform = cbl_params["platform"]
+
+    thread_name = "Thread {}".format(db_name)
+
+    try:
+        repl_obj = Replication(base_url)
+        authenticator = Authenticator(base_url)
+        cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username, ttl=900000)
```

Comment:
Can we define the ttl in separate variable , for readability

Reply:
done